### DISCUSSION TITLE
Could not run the unit test

### DISCUSSION THREAD

**Discussion by laurenceyu2020** [AI4Finance-Foundation/ElegantRL#363] [Q&A] [answered]
Hi:

I tried to run the unit test in VS code. But I got the following errors:

```
2024-07-13 08:31:15.083 [error] Reading directory to watch failed [Error: ENOENT: no such file or directory, scandir '/Users//git/ElegantRL/unit_tests/.pixi/envs'
	at Object.readdirSync (node:fs:1515:3)
	at t.readdirSync (node:electron/js2c/node_init:2:11289)
	at /Users//.vscode/extensions/ms-python.python-2024.10.0-darwin-arm64/out/client/extension.js:2:574598
	at /Users//.vscode/extensions/ms-python.python-2024.10.0-darwin-arm64/out/client/extension.js:2:574789
	at Array.map (<anonymous>)
	at d.initWatchers (/Users/haohaiyu/.vscode/extensions/ms-python.python-2024.10.0-darwin-arm64/out/client/extension.js:2:574507)
	at async d.ensureWatchersReady (/Users/haohaiyu/.vscode/extensions/ms-python.python-2024.10.0-darwin-arm64/out/client/extension.js:2:536827)] {
  errno: -2,
  syscall: 'scandir',
  code: 'ENOENT',
  path: '/Users/haohaiyu/git/ElegantRL/unit_tests/.pixi/envs'
}
```

my python version is
Python 3.10.1

I have installed the required packages.

Anyone know how to fix this? and why there is pixi here which does not mention in the docs.

**Comment by VyvaHart** [comment, marked as answer]
Hi @laurenceyu2020 

Seems like the system can't find directory you provide (ENOENT). Check all paths one more time
if in your case it is not the solution, try:

- open VS settings.json: type _>preferences: open user settings (json)_ in VS code search bar above and edit settings.json file.
you probably will have smth like that:
```
{
  "python.testing.pytestEnabled": true,
  "python.testing.unittestEnabled": false,
  "python.testing.pytestArgs": [
    "unit_tests"
  ],
  "python.envFile": "${workspaceFolder}/unit_tests/.pixi/envs"
}
```
remove **.pixi** in the path ("python.envFile": "${workspaceFolder}/unit_tests/envs") and try to run it again

**or**

if in your case **.pixi** folder is necessary try to create this folder and run script again
`mkdir -p /Users/haohaiyu/git/ElegantRL/unit_tests/.pixi/envs`

good luck bro

**Reply by laurenceyu2020** [reply]
Thank you it worked!

**Reply by VyvaHart** [reply]
Can you then tag my comment as answer, please

**Reply by laurenceyu2020** [reply]
done
